chsdi/templates/htmlpopup/shop_product_group.mako

```diff
@@ -3,24 +3,39 @@
 <%def name="table_body(c, lang)">
 
 <%
-   lang = lang if lang in ('fr','it','en') else 'de'
-   fid = c['featureId']
-   webDavHost = request.registry.settings['webdav_host']
-   image = webDavHost + '/swisstopoproducts/250/' + fid + '.jpg'
-   scale = h.format_scale(c['attributes']['scale'])
-   price = h.format_price(c['attributes']['price'])
-   if c['attributes']['available'] == False :
-     available = _('shop_availability')
-   else :
-     available = ""
+    lang = lang if lang in ('fr','it','en') else 'de'
+    fid = c['featureId']
+    webDavHost = request.registry.settings['webdav_host']
+    image = webDavHost + '/swisstopoproducts/250/' + fid + '.jpg'
+    name = 's_title_%s' % lang
+    if c['attributes']['scale']:
+        c['attributes']['scale'] = h.format_scale(c['attributes']['scale'])
+    if c['attributes']['price']:
+        c['attributes']['price'] = h.format_price(c['attributes']['price'])
+    attr = []
+    attr_poss = ['number', 'name', 'scale', 'release', 'data', 'isbn', 'author', 'price']
+    for ap in attr_poss:
+        if c['attributes'][ap]:
+            attr.append(ap)
+    if c['attributes']['available'] == False:
+        attr.append('available')
+    rowspan = len(attr)
+    html = ''
+    for a in attr:
+        left_col = _('ch.swisstopo.lk25-papierkarte.metadata.%s' % a)
+        value = c['attributes']['%s' % a]
+        if a == 'available':
+            available = _('shop_availability')
+            html += "    <tr><td></td><td valign=\"top\">%s</td></tr>" % available
+        if attr.index(a) == 0:
+            html += "    <tr><td height=10 class=\"cell-left\">%s</td> <td>%s</td> <td rowspan=%s><img src=\"%s\" height=\"150\" width=\"102\" align=\"right\"/></td></tr>" % (left_col, value, rowspan, image)
+        elif attr.index(a) == len(attr) -1 and a != 'available':
+            html += "    <tr><td valign=\"top\" class=\"cell-left\">%s</td> <td valign=\"top\">%s</td></tr>" % (left_col, value)
+        elif a != 'available':
+            html += "    <tr><td height=10 class=\"cell-left\">%s</td> <td>%s</td></tr>" % (left_col, value)
+
 %>
 
-    <tr><td
-    class="cell-left">${_('ch.swisstopo.lk25-papierkarte.metadata.number')}</td>
-    <td>${c['attributes']['number']}</td>       <td rowspan="6"><img src="${image}" height="150" width="102" align="right"/></td></tr> 
-    <tr><td class="cell-left">${_('ch.swisstopo.lk25-papierkarte.metadata.name')}</td>     <td>${c['attributes']['name']}</td></tr>
-    <tr><td class="cell-left">${_('ch.swisstopo.lk25-papierkarte.metadata.scale')}</td>    <td>${scale}</td></tr>
-    <tr><td class="cell-left">${_('ch.swisstopo.lk25-papierkarte.metadata.price')}</td>    <td>${price}</td></tr>
-    <tr><td class="cell-left">${_('ch.swisstopo.lk25-papierkarte.metadata.release')}</td>  <td>${c['attributes']['release']}</td></tr>
-    <tr height=50><td></td><td>${available}</td></tr>
+${html|n}
+
```

Comment:
yes, but I would need to change the name of the attributes in the table, because the layers have a name, but it's also the generic name (used in the legend). Is it possible to add an attribute that does not exist in the model?